google_guest_agent/instance_setup.go

```diff
@@ -157,7 +157,7 @@ func agentInit(ctx context.Context) error {
 					logger.Warningf("Failed to write instance ID file: %v", err)
 				}
 			}
-			if newMetadata.Instance.ID.String() != string(instanceID) {
+			if newMetadata.Instance.ID.String() != strings.TrimSuffix(string(instanceID), "\n") {
```

Comment:
Think about using https://golang.org/pkg/strings/#TrimSpace here, assuming you trust what you get from metadata